build_tools/bazel/iree_bytecode_module.bzl

```
@@ -34,23 +35,34 @@ def iree_bytecode_module(
         linker_tool: the linker to use.
             Defaults to the lld from the llvm-project directory.
         c_identifier: Optional. Enables embedding the module as C data.
+        static_lib_path: When set, the module is compiled into a LLVM static
+            library with the specified library path.
         deps: Optional. Dependencies to add to the generated library.
         **kwargs: any additional attributes to pass to the underlying rules.
     """
 
     if not module_name:
         module_name = "%s.vmfb" % (name)
 
+    out_files = [module_name]
+    module_flags = ["--output-format=vm-bytecode"]
+    if static_lib_path:
+        static_header_path = static_lib_path.replace(".o", ".h")
+        out_files.extend([static_lib_path, static_header_path])
+        module_flags.extend([
+            "--iree-llvm-link-embedded=false",
+            "--iree-llvm-link-static",
+            "--iree-llvm-static-library-output-path=$(location %s)" % (static_lib_path),
+        ])
+
```

Rather than creating a separate variable and then adding it on, just use the single variable "flags" and extend it with the new stuff? I think Starlark does away with some of Python's foot-gun reference semantics so modifying the list passed in shouldn't be a problem (if it is, you can also use `+=` instead of `extend`/`append`). Note that this is slightly different than what you have semantically because it adds the rule's flags at the end instead of the beginning. I'm not sure whether that's practically a difference. If it is, it's probably better to leave as you have it with the user-provided flags at the end


```suggestion
    out_files = [module_name]
    flags.append("--output-format=vm-bytecode")
    if static_lib_path:
        static_header_path = static_lib_path.replace(".o", ".h")
        out_files.extend([static_lib_path, static_header_path])
        flags.extend([
            "--iree-llvm-link-embedded=false",
            "--iree-llvm-link-static",
            "--iree-llvm-static-library-output-path=$(location %s)" % (static_lib_path),
        ])

```

---

buildifier accepts `+=` to replace `extend`, but it still prefers `append` over `+=`. Made the change based on the feedback and the buildifier result.